docs/getting_started.rst

```diff
@@ -0,0 +1,283 @@
+
+Once preloaded, DYAD's C API will intercept the :code:`open` and :code:`fopen` functions when consuming
+files and the :code:`close` and :code:`fclose` functions when producing files. As a result,
+if their code already uses thse functions, users do not need to change their code.
+
+C++ API
+*******
+
+DYAD's C++ API is implemented as a small library that wraps C++'s Standard Library file streams.
+To use DYAD's C++ API, first, add the following to your code:
+
+.. code-block:: cpp
+
+   #include <dyad_stream_api.hpp>
+
+This header defines thin wrappers around the file streams provided by the C++ Standard Library.
+More specifically, it provides the following classes:
+
+* :code:`dyad::basic_ifstream_dyad`
+* :code:`dyad::ifstream_dyad`
+* :code:`dyad::wifstream_dyad`
+* :code:`dyad::basic_ofstream_dyad`
+* :code:`dyad::ofstream_dyad`
+* :code:`dyad::wofstream_dyad`
+* :code:`dyad::basic_fstream_dyad`
+* :code:`dyad::fstream_dyad`
+* :code:`dyad::wfstream_dyad`
+
+When using DYAD, these file streams should be used in place of the file streams from the C++
+Standard Library. The DYAD file streams should be directly used to do the following:
+
+* Open files (with the file stream's :code:`open` method)
+* Close files (with the file stream's :code:`close` method or destructor)
+* Access the underlying C++ Standard Library file stream using the DYAD stream's :code:`get_stream` method
+
+All reading from and writing to files should be done using the underlying C++ Standard Library file stream.
+A simple example of using DYAD's C++ API in a producer application is shown below:
+
+.. code-block:: cpp
+
+   #include <dyad_stream_api.hpp>
+
+   void produce_file(std::string& full_path, int32_t* data, std::size_t data_size)
+   {
+       dyad::ofstream_dyad ofs_dyad;
+       ofs_dyad.open(full_path, std::ofstream::out | std::ios::binary);
+       std::ofstream& ofs = ofs_dyad.get_stream();
+       ofs.write((char*) data, data_size);
+       ofs_dyad.close();
+   }
+
+After replacing C++ Standard Library file streams with their DYAD equivalents,
+there is one final requirement to using the C++ API. When compiling your code,
+you must link the associated library (i.e., :code:`libdyad_stream.so` or
+:code:`libdyad_stream.a`). This library can be found in the :code:`lib`
+subdirectory of the install prefix.
+
+Running DYAD
+############
+
+There are three steps to running DYAD-enabled applications:
+
+1. :ref:`Create a Flux key-value store (KVS) namespace <Create a Flux KVS Namespace>`
+2. :ref:`Determine the managed directories for each application <Determine the Managed Directories for Each Application>`
+3. :ref:`Load DYAD's Flux module <Load DYAD's Flux Module>`
+4. :ref:`Configure and run the DYAD-enabled applications <Configure and Run the DYAD-Enabled Applications>`
+
+Create a Flux KVS Namespace
+***************************
+
+The first step in running DYAD is to create a namespace in the Flux's hierarchical key-value store (KVS).
```

Comment:
Mention that this is to isolate DYAD from the rest of key-value entires used by Flux.